sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Rules/UseUriInsteadOfString.cs

```diff
@@ -56,12 +63,27 @@ protected sealed override void Initialize(SonarAnalysisContext context)
                     var methodDeclaration = (MethodDeclarationSyntax)c.Node;
                     var methodSymbol = c.SemanticModel.GetDeclaredSymbol(methodDeclaration);
 
+                    if (methodSymbol.IsOverride)
+                    {
+                        return;
+                    }
+
                     if (methodSymbol.ReturnType.Is(KnownType.System_String) &&
-                        !methodSymbol.IsOverride &&
                         NameContainsUri(methodSymbol.Name))
                     {
                         c.ReportDiagnostic(Diagnostic.Create(rule_S3995, methodDeclaration.ReturnType.GetLocation()));
                     }
+
+                    foreach (ParameterSyntax param in methodDeclaration.ParameterList.Parameters)
```

Comment:
This is the point I disagree with and think that var hurts code readability.

The type of Parameters property is not clear at this point. For example, how to get the name of the property in these cases?

1. foreach (var param in methodDeclaration.ParameterList.Parameters)
2. foreach (var param in invokedMethod.Parameters)

It could be either param.Identifier.ValueText (1) or param.Name (2). To find out, I'd need to mouse over the var keyword - which is quite annoying.